x-pack/plugins/observability_solution/infra/public/components/asset_details/tabs/dashboards/actions/edit_dashboard.tsx

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { EuiButtonEmpty } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React, { useState } from 'react';
+import type {
+  DashboardItemWithTitle,
+  InfraCustomDashboardAssetType,
+} from '../../../../../../common/custom_dashboards';
+import { SaveDashboardModal } from './save_dashboard_modal';
+
+export function EditDashboard({
+  onRefresh,
+  currentDashboard,
+  assetType,
+}: {
+  onRefresh: () => void;
+  currentDashboard: DashboardItemWithTitle;
+  assetType: InfraCustomDashboardAssetType;
+}) {
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  return (
+    <>
+      <EuiButtonEmpty
+        color="text"
+        size="s"
+        iconType="pencil"
+        data-test-subj="infraEditCustomDashboardMenu"
+        onClick={() => setIsModalVisible(!isModalVisible)}
```

Comment:
nit: should we create a named function for this event? Looks like we're repeating this in the `<SaveDashboardModal>` component